rust/lance-encoding/src/encodings/physical.rs

```diff
@@ -121,5 +122,6 @@ pub fn decoder_from_array_encoding(
         //
         // This will change in the future when we add support for struct nullability.
         pb::array_encoding::ArrayEncoding::Struct(_) => unreachable!(),
+        pb::array_encoding::ArrayEncoding::Binary(_) => unreachable!(),
```

Comment:
We can approach this PR in two ways.  We can wait for the decoder to be added to a single PR before merging or we need some way to add the feature incrementally without breaking the functionality we have.

Reply:
We are doing the former :)